connmgr/connmanager.go

```diff
@@ -202,6 +202,15 @@ func (cm *ConnManager) handleFailedConn(c *ConnReq) {
 		}
 		log.Debugf("Retrying connection to %v in %v", c, d)
 		time.AfterFunc(d, func() {
+
+			// During the time we wait for retry there is a chance that
+			// this connection was already cancelled.
+			if c.State() == ConnCanceled {
+				log.Debugf("Ignoring retry for "+
+					"canceled connreq=%v", c)
+				return
+			}
+
 			cm.Connect(c)
```

Comment:
Do you know why they did not put this condition into `Connect()` and handle it for all cases?

Reply:
Great question. I think that is a better as well. Will change.